mitosheet/src/mito/components/toolbar/GraphTabs/ChartDesignTabContents.tsx

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Mito
+
+
+import React from 'react';
+
+import { GraphParamsBackend, MitoAPI } from '../../..';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import { ActionEnum, AnalysisData, ColumnID, RecursivePartial, SheetData, UIState } from '../../../types';
+import { Actions } from '../../../utils/actions';
+import { updateObjectWithPartialObject } from '../../../utils/objects';
+import Dropdown from '../../elements/Dropdown';
+import DropdownItem from '../../elements/DropdownItem';
+import { convertBackendtoFrontendGraphParams } from '../../taskpanes/Graph/graphUtils';
+import { TaskpaneType } from '../../taskpanes/taskpanes';
+import ToolbarButton from '../ToolbarButton';
+import { AddChartElementButton } from './AddChartElementButton';
+import { ChangeChartTypeButton } from './ChangeChartTypeButton';
+import { GraphTypeConfigurations } from './GraphTypeConfigurations';
+
+export const ChartDesignTabContents = (
+    props: {
+        actions: Actions;
+        uiState: UIState;
+        mitoAPI: MitoAPI;
+        setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+        analysisData: AnalysisData;
+        sheetDataArray: SheetData[];
+        selectedColumnsIds?: ColumnID[];
+        mitoContainerRef: React.RefObject<HTMLDivElement>;
+    }): JSX.Element => {
+    const currOpenTaskpane = props.uiState.currOpenTaskpane;
+    if (currOpenTaskpane.type !== TaskpaneType.GRAPH) {
+        throw new Error('ChartDesignTabContents should only be used when the graph taskpane is open.');
+    }
+    const openGraph = currOpenTaskpane.openGraph;
+    
+    // Get the step params and the step id of the currently open graph by getting the most recent step from this array
```

Comment:
Some documentation to leave:
1. a note how this needs about the live updating hook updating when we make an edit here -- and the hack we use to make this happen
2. This requires the assumption that the open graph has send a message (e.g. this only works because graph automatically refresh)